nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/RestartMonitoring.java

```diff
@@ -0,0 +1,183 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+/**
+ * Asadmin command to restart Module Monitoring Level.
+ * Sets monitoring module levels to OFF before reverting back to their previous values.
+ * 
+ * This was Feature request due to AMX metrics "breaking" sometimes, where restarting it manually was somewhat tedious and prone to error.
+ * 
+ * @author Alan Roth
+ */
+@Service(name = "restart-monitoring")
+@PerLookup
+@ExecuteOn({RuntimeType.DAS, RuntimeType.INSTANCE})
+@TargetType({CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.DEPLOYMENT_GROUP, CommandTarget.CONFIG})
+public class RestartMonitoring implements AdminCommand {
+
+    @Param(name = "target", optional = true, defaultValue = "server-config")
+    String target;
+
+    @Param(name = "verbose", optional = true, shortName = "v", defaultValue = "false")
+    private Boolean verbose;
+
+    @Inject
+    private Target targetUtil;
+
+    @Inject
+    private Logger logger;
+
+    private MonitoringService monitoringService;
+    
+    private ActionReport actionReport;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        actionReport = context.getActionReport();
+
+        List<String> validModuleList = Arrays.asList(Constants.validModuleNames);
+        Config config = targetUtil.getConfig(target);
+        if (config != null) {
+            monitoringService = config.getMonitoringService();
+        } else {
+            actionReport.setMessage("Cound not find target: " + target);
+            actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+
+        Map<String, String> previouslyEnabledModules = getEnabledModules(validModuleList);
+        
+        setModulesToOff(previouslyEnabledModules);
+        setModules(previouslyEnabledModules);
+        
+        if(actionReport.getActionExitCode().equals(actionReport.getActionExitCode().WARNING)){
+            actionReport.appendMessage("Restarted monitoring levels with warnings");
+        }
+        
+        if (verbose) {
+            actionReport.setMessage(getFormattedColumns(previouslyEnabledModules).toString());
+        } else {
+            actionReport.setMessage("Restarted " + previouslyEnabledModules.size() + " modules");
+        }
+    }
+
+    private void setModulesToOff(Map<String, String> modules){
```

Comment:
This should maybe be implemented with the help of `setModules`.

Reply:
Yep